### PR TITLE
Feat/recaps sign in

### LINKED ISSUE #118: Add recaps functionality in notify client js

### PR DESCRIPTION
**Pull request by devceline**
> [!NOTE]
> Dependent on https://github.com/WalletConnect/walletconnect-utils/pull/162

- Add `prepareRegistrationViaRecaps` method

- Resolves #118

### CODE REVIEW

File: packages/notify-client/src/controllers/engine.ts [hunk 1]
```diff
@@ -131,6 +131,30 @@ export class NotifyEngine extends INotifyEngine {
       return this.finishedInitialLoad;
     };
 
+  public prepareRegistrationViaRecaps: INotifyEngine["prepareRegistrationViaRecaps"] =
```
Comment (chris13524) [inline packages/notify-client/src/controllers/engine.ts:134]:
```suggestion
  public prepareRegistrationViaRecaps: INotifyEngine["prepareRegistrationWithRecaps"] =
```

File: packages/notify-client/src/controllers/engine.ts [hunk 2]
```diff
@@ -131,6 +131,30 @@ export class NotifyEngine extends INotifyEngine {
...
+          domain: params.domain,
+          recapObject: {
+            att: {
+              "https://notify.walletconnect.com": params.allApps
+                ? {
+                    "manage/all-apps-notifications": [{}],
+                  }
+                : {
+                    [`manage/${params.domain}-notifications`]: [{}],
+                  },
```
Comment (chris13524) [inline packages/notify-client/src/controllers/engine.ts:147]: Slightly more DRY

```suggestion
              "https://notify.walletconnect.com": "manage/${params.allApps ? "all-apps" : params.domain}-notifications": [{}],
```

**Review by chris13524 — APPROVED** [review]
LGTM. Really should have some new tests though. Also `with` is better name than `via`

File: packages/notify-client/src/controllers/engine.ts (hunk 2 above)
Comment (devceline) [reply packages/notify-client/src/controllers/engine.ts:147]: It's less readable IMO so I'd rather avoid this

**Pull request merged**